pkg/generators/git_test.go

```diff
@@ -0,0 +1,225 @@
+package generators
+
+import (
+	"context"
+	"fmt"
+	argoprojiov1alpha1 "github.com/argoproj-labs/applicationset/api/v1alpha1"
+	argov1alpha1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/reposerver/apiclient"
+	"github.com/argoproj/argo-cd/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+type clientSet struct {
+	RepoServerServiceClient apiclient.RepoServerServiceClient
+}
+
+func (c *clientSet) NewRepoServerClient() (util.Closer, apiclient.RepoServerServiceClient, error) {
+	return util.NewCloser(func() error { return nil }), c.RepoServerServiceClient, nil
+}
+
+type argoCDServiceMock struct {
+	mock.Mock
+}
+
+func (a argoCDServiceMock) GetApps(ctx context.Context, repoURL string, revision string) ([]string, error) {
+	args := a.Called(ctx, repoURL, revision)
+
+	return args.Get(0).([]string), args.Error(1)
+}
+
+func getGitRenderTemplate(name, path string) *argov1alpha1.Application {
+	return &argov1alpha1.Application{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "namespace",
+			Finalizers: []string{
+				"resources-finalizer.argocd.argoproj.io",
+			},
+		},
+		Spec: argov1alpha1.ApplicationSpec{
+			Source: argov1alpha1.ApplicationSource{
+				RepoURL:        "RepoURL",
+				Path:           path,
+				TargetRevision: "HEAD",
+			},
+			Destination: argov1alpha1.ApplicationDestination{
+				Server:    "server",
+				Namespace: "destinationNamespace",
+			},
+			Project: "project",
+		},
+	}
+}
+
+func TestGitGenerateApplications(t *testing.T) {
+	cases := []struct {
+		name		  string
+		template      argoprojiov1alpha1.ApplicationSetTemplate
+		Directories   []argoprojiov1alpha1.GitDirectoryGeneratorItem
+		repoApps      []string
+		repoError     error
+		expected      []argov1alpha1.Application
+		expectedError error
+	}{
+		{
+			"happy flow",
+			argoprojiov1alpha1.ApplicationSetTemplate{
```

Comment:
Is this template the same in each test? If so pulling it out would make the tests a lot more concise. 

If it is differing a function with params might be a good idea for generating it and then reducing each test to one line.